### PR TITLE
Added test that no files are executable

### LINKED ISSUE #14845: Add test that no files are executable
In https://github.com/sympy/sympy/pull/14844 I fixed an issue where some files in the vector module were had the executable bit set. We should add a test that runs in Travis that goes through all the files in git and makes sure none have an executable bit set, unless they are in a whitelist (anything in `bin/` should be whitelisted, `setup.py`, a few scripts in `release/`, maybe something else). You can use `git ls-files -s` to list the files in git with their permissions.

It should be a new script in `bin` that runs in one of the Travis builds.

### PR DESCRIPTION
**Pull request by ramvenkat98**
#### References to other Issues or PRs

Fixes #14845 

#### Brief description of what is fixed or changed
Created test_executable.py which tests that only files with shebangs have
an executable bit set. These two files violated this and had permissions
changed (to code number 644):
    sympy/utilities/tests/test_module_imports.py
    data/TeXmacs/progs/init-sympy.scm

#### Release Notes

NO ENTRY

### CODE REVIEW

File: bin/test_executable.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+"""
```
Comment (asmeurer) [inline bin/test_executable.py]: Make this file itself executable.

**Comment by asmeurer** [comment]
You should add this to the Travis tests. The best place would be [here](https://github.com/sympy/sympy/blob/b70b86b2b948f500782b38f2a20524aa2205a815/bin/test_travis.sh#L36).

File: bin/test_executable.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+from get_sympy import path_hack
+base_dir = path_hack()
+
+def test_executable(path):
+    if not os.path.isdir(path):
+        if os.access(path, os.X_OK):
+            with open(path, 'r') as f:
+                if f.readline()[:2] != "#!":
+                    exn_msg = "Executable bit set for file path" + path
```
Comment (asmeurer) [inline bin/test_executable.py]: Note in the error that the file either should not be executable, or it should have a shebang line.

**Comment by ramvenkat98** [comment]
I've made the modifications suggested by the review and updated the pull request, namely:

1. Making the test file executable by adding a shebang line and setting the permission to 755
2. Added the test file to Travis builds
3. Made the exception message clearer
